qiskit/circuit/register.py

```diff
@@ -160,7 +161,7 @@ def __getitem__(self, key):
             if max(key) < len(self):
                 return [self._bits[idx] for idx in key]
             else:
-                raise CircuitError("register index out of range")
+                raise QiskitIndexError("register index out of range")
```

Comment:
This is a breaking API change for anyone trying to catch this exception from getitem and we should not do this. Instead just update the documentation to reflect what the api actually does.

Reply:
changing to fix the docstring